cmd/kube-controller-manager/app/plugins.go

```diff
@@ -69,6 +76,31 @@ func ProbeRecyclableVolumePlugins(flags VolumeConfigFlags) []volume.VolumePlugin
 	return allPlugins
 }
 
+// NewVolumeProvisionersForCloud maps a cloud provider to a specific volume plugin.
+func NewVolumeProvisionersForCloud(cloud cloudprovider.Interface) map[string]volume.ProvisionableVolumePlugin {
+	var provisioner volume.VolumePlugin
+	switch {
+	case cloud != nil && cloud.ProviderName() == aws_cloud.ProviderName:
+		provisioner = aws_ebs.ProbeVolumePlugins()[0]
+	case cloud != nil && cloud.ProviderName() == gce_cloud.ProviderName:
+		provisioner = gce_pd.ProbeVolumePlugins()[0]
+	case cloud != nil && cloud.ProviderName() == openstack.ProviderName:
+		provisioner = cinder.ProbeVolumePlugins()[0]
+	}
+
+	plugins := map[string]volume.ProvisionableVolumePlugin{
+		"experimental-hostpath-testing-only": host_path.ProbeVolumePlugins(volume.VolumeConfig{})[0].(volume.ProvisionableVolumePlugin),
+	}
+
+	if provisioner != nil {
+		plugins["gold"] = provisioner.(volume.ProvisionableVolumePlugin)
```

Comment:
Seems like we should define a config for this and move the assumptions about cloud providers into the appropriate place.